Helper/Session.php

```diff
@@ -167,4 +175,26 @@ public function loadSession($quote)
         }
         $this->replaceQuote($quote);
     }
+
+    /**
+     * Load and set cached form key
+     *
+     * @param Quote $quote
+     */
+    public function setFormKey($quote)
+    {
+        $cacheIdentifier = self::BOLT_SESSION_PREFIX_FORM_KEY . $quote->getId();
+        $this->formKey->set($this->cache->load($cacheIdentifier));
+    }
+
+    /**
+     * Store form key in cache for use later from the create order API call
+     *
+     * @param Quote $quote
+     */
+    public function cacheFormKey($quote)
+    {
+        $cacheIdentifier = self::BOLT_SESSION_PREFIX_FORM_KEY . $quote->getId();
+        $this->cache->save($this->formKey->getFormKey(), $cacheIdentifier, [], 86400);
```

Comment:
@YevhenShBolt if you try bolt-preauth backend-order on Magento 2.3.x you will see the redirect URL goes into a infinite redirect loop for a while until the server displays and error. The correct needs to have a correct key segment, like this one magento2.guaranteed.site/admin/sales/order/index/key/03064edfb913392c4d570096905ccde02e00cd0398d497e8d7cc1b80163ee1f7/.
When this URL is created the form_key is used to generate the key. When Bolt make create order REST API call, we send the redirect URL in result. For Magento to generate the valid URL it needs form_key or it will create a new one and create the URL invalid to the session. So we need to get out session form_key and keep it somewhere where Create Order can find it. The same as we did for backend / frontend order flag, which we stored in the order table. I wanted to avoid another DB field so I used the cache.
I agree 1 day is too long. 1 hour should be enough but I will make it 4 just in case.

Reply:
i agree